### PR TITLE
add branch to clone command

### PR DESCRIPTION
**Pull request by johnpryan**

### CODE REVIEW

**Comment by dmathieu** [comment]
I believe we need to default it to master.
Or maybe even not specify any branches if the value isn't there, so it still works if the default branch isn't master.

**Comment by johnpryan** [comment]
@dmathieu f9c38b517eb5231173779dfdce091c2f63c4c69e should address your concerns, I'll try to test it ASAP

**Comment by johnpryan** [comment]
I went with the second approach (not specifying a branch in the command unless the configuration has a 'branch' key)

File: bin/install_git_submodules.rb [hunk 1]
```diff
@@ -5,10 +5,16 @@
 config.get_params.each do |param|
   next unless param.match(/^submodule/)
   c = config[param]
+  
+  branch = c["branch"]
 
   puts "---> Installing submodule #{c["path"]}"
-  `git clone #{c["url"]} #{ENV['BUILD_DIR']}/#{c["path"]}`
-
+  if branch
```
Comment (dmathieu) [inline bin/install_git_submodules.rb]: Fully duplicating the command here isn't awesome. How about:

``` ruby
branch_flag = c["branch"] ? "-b #{c['branch']}" : ""
`git clone #{c["url"]} #{branch_flag} #{ENV['BUILD_DIR']}/#{c["path"]}`
```

Comment (johnpryan) [reply bin/install_git_submodules.rb]: good idea - fixed

**Comment by dmathieu** [comment]
Thank you!

**Pull request merged**